engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -202,42 +202,42 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
       case e: OperatorPower => checkTwoOperandsArithmeticOperation(validationContext, e, current)(NumberTypesPromotionStrategy.ForPowerOperation)
 
       case e: OpPlus => withTypedChildren {
-        case left :: right :: Nil if left == Unknown || right == Unknown => Valid(Unknown)
-        case left :: right :: Nil if left.canBeSubclassOf(Typed[String]) || right.canBeSubclassOf(Typed[String]) => Valid(Typed[String])
-        case left :: right :: Nil if left.canBeSubclassOf(Typed[Number]) && right.canBeSubclassOf(Typed[Number]) => Valid(commonSupertypeFinder.commonSupertype(left, right)(NumberTypesPromotionStrategy.ForMathOperation))
+        case left :: right :: Nil if left.typingResult == Unknown || right.typingResult == Unknown => Valid(TypingResultWithContext(Unknown))
```

Comment:
We can do extraction in such cases to make less invocations of `.typingResult`? I mean:
`case TypingResultWithContext(left, _) :: TypingResultWithContext(right, _) :: Nil if left == Unknown ... `

Reply:
changed it in most places, looks better, thanks